eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -53,15 +57,31 @@ case class AuditResponse(sent: Seq[PaymentSent], received: Seq[PaymentReceived],
 
 case class TimestampQueryFilters(from: Long, to: Long)
 
+case class MutualCloseStatus(unpublished: Satoshi, unconfirmed: Satoshi, confirmed: Satoshi)
+
+case class GlobalBalance private(total: Btc, onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus)
+
+object GlobalBalance {
+
+  private def apply(total: Btc, onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus): GlobalBalance = ???
+
+  def apply(onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus): GlobalBalance = new GlobalBalance(
+    total = onchain.total + offchain.total,
+    onchain = onchain,
+    offchain = offchain,
+    mutualClose = mutualClose
+  )
+}
```

Comment:
Because that would fail at runtime and I wanted something that fails at compilation.

Reply:
But right now what you have fails at runtime as well since you have an `apply(...) = ???`...
Or is it a leftover that you wanted to remove?

Reply:
The method is defined as `private` the goal was to override the default one and make it inaccessible from the outside.

Reply:
But why can't you simply remove it?

Reply:
It's a default method for case classes. Really I think the problem here is with serialization, I can't believe there is no way of simply adding a field to the base serialization.

Reply:
I agree, I find our serialization code unsatisfying, but maybe it's simply because we haven't spent enough time improving it and use the library imperfectly? It's always been a part of the code we haven't bothered changing much

Reply:
:nerd_face:

```scala
case class GlobalBalance (onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus) {
  val total: Btc = onchain.total + offchain.total
}

class GlobalBalanceSerializer extends CustomSerializer[GlobalBalance](_ => ( {
  null
}, {
  case o: GlobalBalance =>
    JObject(JField("total", JDecimal(o.total.toDouble))) merge Extraction.decompose(o)(DefaultFormats + new BtcSerializer)
}))
```

Reply:
Done with 635b2ce.